core/embed/rust/src/ui/model_tr/component/button_controller.rs

```diff
@@ -105,6 +110,13 @@ where
 {
     pos: ButtonPos,
     button_type: ButtonType<T>,
+    /// Holds the timestamp of when the button was pressed.
+    pressed_since: Option<Instant>,
+    /// How long the button should be pressed to send `long_press=true` in
+    /// `ButtonControllerMsg::Triggered`
+    long_press_ms: u32,
+    /// Timer for sending `ButtonControllerMsg::LongPressed`
+    long_pressed_timer: Option<TimerToken>,
```

Comment:
Nit: it would kinda make sense to keep both in a single option, i.e. `Option<(Instant, TimerToken)>` since they're not really independent.